src/coprocessor/dag/expr/builtin_math.rs

```diff
@@ -361,6 +361,25 @@ impl ScalarFunc {
         };
         Ok(Some(r))
     }
+
+    #[inline]
+    pub fn truncate_decimal(
+        &self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<Decimal>>> {
+        let x = try_opt!(self.children[0].eval_decimal(ctx, row));
+        let d = try_opt!(self.children[1].eval_int(ctx, row));
+        let d = if mysql::has_unsigned_flag(self.children[1].get_tp().get_flag()) {
+            (d as u64).min(127) as i8
```

Comment:
Why do you choose 127 here? Generally, mysql may use 30 as the threshold. You can check on [https://www.w3schools.com/sql/trymysql.asp?filename=trysql_func_mysql_truncate](https://www.w3schools.com/sql/trymysql.asp?filename=trysql_func_mysql_truncate)

Reply:
127(i8::max_value) is the maximum i8 value,  I use 127( and -128 below) just to avoid the truncation where cast i64/u64 to i8. It is not intent to be the  max acceptable value as the threshold.
IMHO, the threshold 30 is already sit in the decimal.rs (MAX_FRACTION), and the Decimal#round will handle it (https://github.com/tikv/tikv/blob/master/src/coprocessor/codec/mysql/decimal.rs#L1154). so in here ,we just need a `valid` i8 vlaue.